wagtail/test/utils/wagtail_tests.py

```diff
@@ -1,12 +1,43 @@
 import warnings
 from contextlib import contextmanager
 
+from bs4 import BeautifulSoup
 from django import VERSION as DJANGO_VERSION
 from django.contrib.auth import get_user_model
+from django.test.client import Client as DjangoTestClient
 from django.test.testcases import assert_and_parse_html
 
 
+class BS4SoupClient(DjangoTestClient):
+    """Convenience wrapper around Django's Client and BeautifulSoup's SoupSieve
+
+    The intend of this wrapper is to make it easier to check the html of a response.
+
+    Usage example:
+
+        client = BS4SoupClient()
+        response = client.get("/good-vibes/")
+
+        # Select a HTML <span class="good-vibes> in the response
+        good_vibes = response.select("span.good-vibes")
+
+        # Assert the is only one element on the page that matches the selector
+        self.assertEqual(len(soup), 1)
+    """
+
+    def request(self, **request):
+        response = super().request(**request)
+        response.soup = BeautifulSoup(response.content, "html5lib")
```

Comment:
Question: What happens if the content isn't HTML?

Issue: I believe a lot of computation is done when constructing a "soup". Perhaps we should defer this until the soup is actually used. I could imagine this slowing down tests quite a lot.

Reply:
>Question: What happens if the content isn't HTML?

Good point! I added a check for this. Let me know if that looks okay to you.

> Issue: I believe a lot of computation is done when constructing a "soup". Perhaps we should defer this until the soup is actually used. I could imagine this slowing down tests quite a lot.

I had not considered this at all. Thanks for raising! 

I changed this and wrapped the soup with `django.utils.functional.SimpleLazyObject` which instantiates the soup upon initial access and memoizes it for further operations.

Happy to hear your thoughts on my changes @RealOrangeOne